docs/design/assetgeneration.md

```diff
@@ -61,7 +64,7 @@ The installer uses depth-first traversal on the dependency graph, starting at th
 
 ### Dirty detection
 
-An asset generation reports **DIRTY** when it detects that the components have been modified from previous run.
+An asset generation reports **DIRTY** when it detects that the components have been modified from previous run, for now the asset is considered dirty when it's on-disk.
 
 When generating dependencies of an asset, if any one of the dependencies report dirty, the installer informs the asset that its dependencies are dirty. The asset can either generate from its dependencies or exit with error.
```

Comment:
This part is not true. Do we have plans to inform the asset and let it make a decision? Or should we remove this part?

Reply:
removed.